src/audio/codec_adapter/codec_adapter.c

```diff
@@ -311,14 +311,23 @@ int codec_adapter_params(struct comp_dev *dev,
 		rfree(mod->stream_params);
 
 	mod->stream_params = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
-				     sizeof(*mod->stream_params));
+				     sizeof(*mod->stream_params) + params->ext_data_length);
 	if (!mod->stream_params)
 		return -ENOMEM;
 
 	ret = memcpy_s(mod->stream_params, sizeof(struct sof_ipc_stream_params),
 		       params, sizeof(struct sof_ipc_stream_params));
 	assert(!ret);
 
+	if (params->ext_data_length) {
+		ret = memcpy_s((uint8_t *)mod->stream_params +
+			       sizeof(struct sof_ipc_stream_params),
+			       params->ext_data_length,
+			       (uint8_t *)params + sizeof(struct sof_ipc_stream_params),
+			       params->ext_data_length);
+		assert(!ret);
```

Comment:
Can we return an error instead?

Reply:
The pre-existing code without ext_data_length uses assert. I can do that in a separate patch for both. Or remove the assert in the first place.

This is pre-existing code:
```
       ret = memcpy_s(&mod->stream_params, sizeof(struct sof_ipc_stream_params),
»       »              params, sizeof(struct sof_ipc_stream_params));
»       assert(!ret);

```

Reply:
yes, lets remove the assert please. I think we're doing that elsewhere in the FW code too

Reply:
yeah, lets return an error. I would imagine integrators would be using this code path a lot and frequently sending different configs as part of development.